File: presentation/src/main/java/com/unibs/zanotti/inforinvestigador/comments/ExpandableCommentGroup.java
package com.unibs.zanotti.inforinvestigador.comments;

import com.unibs.zanotti.inforinvestigador.domain.model.Comment;
import com.xwray.groupie.ExpandableGroup;

public class ExpandableCommentGroup extends ExpandableGroup {
    public ExpandableCommentGroup(Comment comment, int depth, ExpandableCommentItem.OnReplyClickedListener replyListener) {
        super(new ExpandableCommentItem(comment, depth, replyListener));

        for (Comment comm : comment.getChildren()) {
            add(new ExpandableCommentGroup(comm, ++depth, replyListener));
        }
    }
}

File: app/src/main/java/com/unibs/zanotti/inforinvestigador/comments/ExpandableCommentGroup.kt
package com.unibs.zanotti.inforinvestigador.comments

import com.unibs.zanotti.inforinvestigador.data.Comment
import com.xwray.groupie.ExpandableGroup

class ExpandableCommentGroup constructor(comment: Comment, depth: Int = 0) :
    ExpandableGroup(ExpandableCommentItem(comment, depth)) {

    init {
        for (comm in comment.children) {
            add(ExpandableCommentGroup(comm, depth.plus(1)))
        }
    }
}